src/cpp/benders/benders_core/BendersBase.cpp

```diff
@@ -9,18 +9,31 @@
 #include "LastIterationReader.h"
 #include "LastIterationWriter.h"
 #include "LogUtils.h"
+#include "VariablesGroup.h"
 #include "glog/logging.h"
 #include "solver_utils.h"
 
-BendersBase::BendersBase(BendersBaseOptions options, Logger logger,
+BendersBase::BendersBase(const BendersBaseOptions &options, Logger logger,
                          Writer writer,
                          std::shared_ptr<MathLoggerDriver> mathLoggerDriver)
-    : _options(std::move(options)),
+    : _options(options),
       _csv_file_path(std::filesystem::path(_options.OUTPUTROOT) /
                      (_options.CSV_NAME + ".csv")),
       _logger(std::move(logger)),
       _writer(std::move(writer)),
-      mathLoggerDriver_(mathLoggerDriver) {}
+      mathLoggerDriver_(std::move(mathLoggerDriver)) {
+  if (options.EXTERNAL_LOOP_OPTIONS.DO_OUTER_LOOP) {
+    //TODO maybe the input format will change?
+    outer_loop_input_data_ =
+        Outerloop::OuterLoopInputFromYaml().Read(OuterloopOptionsFile());
+    outer_loop_biLevel_ = OuterLoopBiLevel(outer_loop_input_data_);
+  }
+}
+
+std::filesystem::path BendersBase::OuterloopOptionsFile() const {
+  return std::filesystem::path(
+         _options.EXTERNAL_LOOP_OPTIONS.OUTER_LOOP_OPTION_FILE);
```

Comment:
I am not convinced with this last change as for the structure file, we access with `options.INPUTROOT / options.STRUCTURE`. Why not keeping the same pattern for the outer loop config file.